### PR TITLE
gh-106831: Fix `NULL` check of `d2i_SSL_SESSION` result in `_ssl.c`

### PR DESCRIPTION
**Pull request by sobolevn**
@vstinner you've recently commited to this module, would you have time to take a look at the fix I am proposing? :)

* Issue: gh-106831

### CODE REVIEW

File: Modules/_ssl.c [hunk 1]
```diff
@@ -2822,7 +2822,8 @@ _ssl_session_dup(SSL_SESSION *session) {
     }
     const_p = senc;
     newsession = d2i_SSL_SESSION(NULL, &const_p, slen);
-    if (session == NULL) {
+    if (newsession == NULL) {
+        PyErr_SetString(PyExc_ValueError, "d2i() failed.");
```
Comment (vstinner) [inline Modules/_ssl.c]: Error messages in Python don't have a trailing dot: you should remove it from your added error message, and the two error messages above.

**Review by vstinner — COMMENTED** [review]
> @vstinner you've recently commited to this module, would you have time to take a look at the fix I am proposing? :)

Oh no. What did I do?

**Comment by sobolevn** [comment]
> Oh no. What did I do?

😆 https://github.com/python/cpython/commit/ad95c7253a70e559e7d3f25d53f4772f28bb8b44

Done, I've addressed your review. Thanks for the quick feedback 👍

File: Modules/_ssl.c [hunk 2]
```diff
@@ -2817,12 +2817,13 @@ _ssl_session_dup(SSL_SESSION *session) {
...
     p = senc;
     if (!i2d_SSL_SESSION(session, &p)) {
-        PyErr_SetString(PyExc_ValueError, "i2d() failed.");
+        PyErr_SetString(PyExc_ValueError, "i2d() failed");
         goto error;
     }
     const_p = senc;
     newsession = d2i_SSL_SESSION(NULL, &const_p, slen);
-    if (session == NULL) {
+    if (newsession == NULL) {
```
Comment (vstinner) [inline Modules/_ssl.c:2825]: Oh, now I see the typo! Previous, *session* was checked, but **new**session must checked! Nicely spotted.

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**